hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/QueryCacheEndToEndProvider.java

```diff
@@ -18,93 +18,98 @@
 
 import com.hazelcast.util.ConstructorFunction;
 
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.concurrent.ConcurrentMap;
 
 import static com.hazelcast.map.impl.querycache.subscriber.NullQueryCache.NULL_QUERY_CACHE;
-import static com.hazelcast.util.ConcurrencyUtil.getOrPutIfAbsent;
-import static com.hazelcast.util.ConcurrencyUtil.getOrPutSynchronized;
 
 /**
  * Provides construction of whole {@link com.hazelcast.map.QueryCache QueryCache}
- * sub-system. As a result of that construction we can have a ready to use {@link com.hazelcast.map.QueryCache QueryCache}.
+ * sub-system. As a result of that construction, we can have a ready to use {@link com.hazelcast.map.QueryCache QueryCache}.
  */
 public class QueryCacheEndToEndProvider<K, V> {
 
     private static final int MUTEX_COUNT = 16;
     private static final int MASK = MUTEX_COUNT - 1;
 
-    private final ConstructorFunction<String, ConcurrentMap<String, InternalQueryCache<K, V>>> constructorFunction
-            = new ConstructorFunction<String, ConcurrentMap<String, InternalQueryCache<K, V>>>() {
-        @Override
-        public ConcurrentMap<String, InternalQueryCache<K, V>> createNew(String arg) {
-            return new ConcurrentHashMap<String, InternalQueryCache<K, V>>();
-        }
-    };
-
     private final Object[] mutexes;
-    private final ConcurrentMap<String, ConcurrentMap<String, InternalQueryCache<K, V>>> queryCaches;
+    private final Map<String, Map<String, InternalQueryCache<K, V>>> mapName2QueryCaches;
```

Comment:
minor: I 'm not a big fan of using "2" as "to", maybe rename map as `queryCachesByMapName`?

Reply:
Changed to use `to`. This parallelism with its key&value nature feels easy to read.